packages/dds/tree/src/test/feature-libraries/flex-tree/utils.ts

```diff
@@ -15,24 +15,26 @@ import {
 // eslint-disable-next-line import/no-internal-modules
 import { Context, getTreeContext } from "../../../feature-libraries/flex-tree/context.js";
 import {
-	DefaultEditBuilder,
 	FlexAllowedTypes,
 	FlexFieldKind,
 	FlexTreeSchema,
 	createMockNodeKeyManager,
 	nodeKeyFieldKey,
 } from "../../../feature-libraries/index.js";
-import { TreeContent } from "../../../shared-tree/index.js";
+import {
+	TreeContent,
+	type ISharedTreeEditor,
+	type ITreeCheckout,
+} from "../../../shared-tree/index.js";
 import { brand } from "../../../util/index.js";
 import { forestWithContent } from "../../utils.js";
 
 export function getReadonlyContext(forest: IEditableForest, schema: FlexTreeSchema): Context {
 	// This will error if someone tries to call mutation methods on it
-	const dummyEditor = {} as unknown as DefaultEditBuilder;
+	const dummyEditor = {} as unknown as ISharedTreeEditor;
 	return getTreeContext(
 		schema,
-		forest,
-		dummyEditor,
+		{ forest, editor: dummyEditor } as unknown as ITreeCheckout,
```

Comment:
My preference for these places would be to create a `MockTreeCheckout` that has only a forest and (maybe automatically) a dummyEditor. Then implement all the properties/functions on the checkout with error messages like "[API] is not available on a mock checkout". A little bit of one-time boilerplate but if anybody ever uses something on the checkout that they can't, they'll immediately know what's wrong in their test rather than just getting a "can't do stuff with undefined" error.

What do you think?

Reply:
Yep, seems reasonable. Implementing.